tests/base/src/test/java/org/wildfly/security/auth/FileSystemSecurityRealmTest.java

```diff
@@ -102,6 +127,21 @@ public void testCreateIdentityWithLevels() throws Exception {
         identity.dispose();
     }
 
+    @Test
+    public void testCreateIdentityWithLevelsEncryption() throws Exception {
+        FileSystemSecurityRealm securityRealm = FileSystemSecurityRealm.builder()
+                .setRoot(getRootPath(false))
+                .setLevels(3)
+                .setSecretKey(key)
+                .build();
+        ModifiableRealmIdentity identity = securityRealm.getRealmIdentityForUpdate(new NamePrincipal("plainUser"));
+
```

Comment:
Should there be a check to make sure the identity doesn't exist here before the call to `create`?